src/audio/tdfb/tdfb_direction.c

```diff
@@ -125,7 +125,9 @@ static int16_t max_mic_distance(struct tdfb_comp_data *cd)
 
 	/* Max lag based on largest array dimension. Microphone coordinates are Q4.12 meters */
 	for (i = 0; i < cd->config->num_mic_locations; i++) {
-		/* Start from i+1 halves the amount of iteration & to eliminate redundant checks */
+		/* Calculate distances between all possible microphone pairs to
+		 * find the maximum distance
+		 */
```

Comment:
for the future: please make corrections, addressing review comments *in the commit where that change is made* - not in a later commit. It doesn't matter much in this case, since this is a comment, but it would make reviewing easier

Reply:
Just FYI @lyakh: using new commits to address review comments is the "standard" GitHub process. Using Github like plain git is very unusual, SOF (and Zephyr) are among the outliers rewriting history and using "force" pushes.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

https://github.com/zephyrproject-rtos/zephyr/issues/39194

Just a warning that this issue will keep coming again and again.

Reply:
@marc-hb do I understand you correctly, that as soon as I push a PR with whatever bugs in my commits, the *standard* / recommended way to improve that PR would be to push fixes on top? So that after a merge you get my original 5 commits with all the bugs in them and then additional 3 commits on top with fixes? If so, I strongly disagree with this process. For one it would break bisection. And in general that isn't what a peer review process is about IMHO. Recall the "original LKML" (at least as of the git era) style review process - you send an email thread with your commits, you get a ton of replies explaining how wrong you are and suggesting all the therapists you should visit to address your problems, after which you go, fix your stuff and resend the emails - the original ones are gone, no fixing on top.
But importantly - we all make mistakes, very few PRs are recognised as good enough in their first version.

Reply:
Agree with @lyakh, This comment change should really be in the previous commit that just added it. It's not much work to shuffle changes from commit to other and @ShriramShastry should take the time to learn to clean up patches this way and ease our review work.

Reply:
I think you missed the last sentence:

> Just a warning that this issue will keep coming again and again.

Reply:
Just for fun, here's a very curious and much more extreme view: https://fossil-scm.org/home/doc/trunk/www/rebaseharm.md